pkg/hub/controllers/cluster/deregister.go

```diff
@@ -0,0 +1,330 @@
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func constructServiceAccount() *corev1.ServiceAccount {
+	svcAcc := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccountName,
+			Namespace: ControlPlaneNamespace,
+		},
+	}
+	return svcAcc
+}
+
+func (r *Reconciler) getOperatorClusterRole(ctx context.Context) (*rbacv1.ClusterRole, error) {
+	operatorClusterRole := &rbacv1.ClusterRole{}
+	err := r.MeshClient.Get(ctx, types.NamespacedName{Name: operatorClusterRoleName}, operatorClusterRole)
+	if err != nil {
+		return nil, err
+	}
+	return operatorClusterRole, nil
+}
+
+func constructClusterRole(clusterRole *rbacv1.ClusterRole) *rbacv1.ClusterRole {
+	clusterRoleCopy := &rbacv1.ClusterRole{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{Name: clusterRoleName, Namespace: ControlPlaneNamespace},
+		Rules:      clusterRole.Rules,
+	}
+
+	return clusterRoleCopy
+}
+
+func constructClusterRoleBinding() *rbacv1.ClusterRoleBinding {
+	clusterRole := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterRoleBindingName,
+			Namespace: ControlPlaneNamespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     clusterRoleName,
+		},
+		Subjects: []rbacv1.Subject{{
+			Kind:      "ServiceAccount",
+			Name:      serviceAccountName,
+			Namespace: ControlPlaneNamespace,
+		}},
+	}
+	return clusterRole
+}
+
+func getConfigmapScriptPath(file string) string {
+	dir := os.Getenv("SCRIPT_PATH")
+	if dir != "" {
+		return path.Join(dir, file+".sh")
+	}
+	return path.Join("../../../../scripts", file+".sh")
+}
+
+func getConfigmapData() (string, error) {
+	fileName := "cleanup"
+	data, err := os.ReadFile(getConfigmapScriptPath(fileName))
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+func constructConfigMap(data string) *corev1.ConfigMap {
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterDeregisterConfigMap,
+			Namespace: ControlPlaneNamespace,
+		},
+		Data: map[string]string{
+			"kubeslice-cleanup.sh": data,
+		}}
+	return cm
+}
+
+func constructJobForClusterDeregister() *batchv1.Job {
+	backOffLimit := int32(0)
+	ttlSecondsAfterFinished := int32(600)
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deregisterJobName,
+			Namespace: ControlPlaneNamespace,
+			Labels: map[string]string{
+				"kubeslice-manager/api-gateway": "cluster-deregister-job",
+			},
+		},
+		Spec: batchv1.JobSpec{
+			BackoffLimit:            &backOffLimit,
+			TTLSecondsAfterFinished: &ttlSecondsAfterFinished,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: deregisterJobName,
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: serviceAccountName,
+					Containers: []corev1.Container{{
+						Name:  serviceAccountName,
+						Image: "aveshadev/worker-installer:latest",
```

Comment:
Is this image going to constant both in dev and prod? OSS and ENT?

Reply:
Yes it is an opensource image.

Reply:
Is this pointing to dev or Prod repo?

Reply:
Sorry I didn't get your point. This is an opensource docker image publicly accessible. This only contains an alpine image, with helm, kubectl and some general utilities jq and yq.